internal/controllers/machine/machine_controller_noderef_test.go

```diff
@@ -784,6 +784,115 @@ func TestGetManagedLabels(t *testing.T) {
 	}
 }
 
+func TestGetManagedAnnotations(t *testing.T) {
+	machine := newFakeMachine("default", "test-cluster")
+	ms := builder.MachineSet("default", "ms").Build()
+	ref := metav1.NewControllerRef(ms, ms.GroupVersionKind())
+
+	defaultAnnotations := map[string]string{
+		clusterv1.ClusterNameAnnotation:      machine.Spec.ClusterName,
+		clusterv1.ClusterNamespaceAnnotation: machine.GetNamespace(),
+		clusterv1.MachineAnnotation:          machine.Name,
+	}
+
+	additionalAnnotations := map[string]string{
+		"foo":                               "bar",
+		"bar":                               "baz",
+		"company.xyz/node.cluster.x-k8s.io": "not-managed",
+		"gpu-node.cluster.x-k8s.io":         "not-managed",
+		"company.xyz/node-restriction.kubernetes.io": "not-managed",
+		"gpu-node-restriction.kubernetes.io":         "not-managed",
+		"wrong.test.foo.com":                         "",
+	}
```

Comment:
```suggestion
	additionalAnnotations := map[string]string{
		"foo":                               "bar",
		"bar":                               "baz",
		"example.com/node.cluster.x-k8s.io": "not-managed",
		"gpu-node.cluster.x-k8s.io":         "not-managed",
		"example.com/node-restriction.kubernetes.io": "not-managed",
		"gpu-node-restriction.kubernetes.io":         "not-managed",
		"wrong.test.example.com":                         "",
	}
```

nits: I'm not exactly sure about the policy of kubernetes community, but `company.xyz` domain is not owned by kubernetes community, which we should avoid for use even in the test cases. I would prefer using `example.com`, or we could use some TLDs for testing purpose, for example, `.test`.

Reply:
I'm fine with that, though I'll note that we don't control `example.com`, either :)

I can edit these, though. Doesn't seem like a big deal.